posts/2022-07-25-keyword-generics.md

```diff
@@ -0,0 +1,343 @@
+---
+layout: post
+title: "Announcing the Keyword Generics Initiative"
+author: Yoshua Wuyts, on behalf of the Keyword Generics Initiative
+release: false
+---
+
+We ([Oli], [Niko], and [Yosh]) are excited to announce the start of the [Keyword
+Generics Initiative][kwi], a new initiative [^initiative] under the purview of
+the language team. We're officially just a few weeks old now, and in this post
+we want to briefly share why we've started this initiative, and share some
+insight on what we're about.
+
+[Oli]: https://github.com/oli-obk
+[Niko]: https://github.com/nikomatsakis
+[Yosh]: https://github.com/yoshuawuyts
+[kwi]: https://github.com/rust-lang/keyword-generics-initiative
+
+[^initiative]: Rust governance terminology can sometimes get confusing. An
+"initiative" in Rust parlance is different from a "working group" or "team".
+Initiatives are intentionally limited: they exist to explore, design, and
+implement specific pieces of work - and once that work comes to a close, the
+initiative will wind back down. This is different from, say, the lang team -
+which essentially carries a `'static` lifetime - and whose work (ominously) does
+not have a clearly defined end.
+
+## A missing kind of generic
+
+One of Rust's defining features is the ability to write functions which are
+_generic_ over their input types. That allows us to write a function once,
+leaving it up to the compiler to generate the right implementations for us.
+
+But while we're able to be generic over _types_ in a function, we're not able to
```

Comment:
This paragraph seems to jump from "being able to be generic over types" to "being generic over asyncness" which is a bit confusing. It seems like you're describing what this feature *isn't* then an *example* of what this feature i. Yet you never attempt to define what the feature itself is. 

It would be nice to say something to the effect of "Rust allows you to be generic over types - it does not allow you to be generic over other things that are usually specified by keywords in Rust. For example, whether a function is async, whether a function can fail or not, whether a function is const or not, etc."

Reply:
Good call; thank you!